sql/parser/builtins.go

```diff
@@ -761,3 +773,73 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+// Round rounds x to n digits.
+// Directly taken from CPython's `double_round` in
+// cpython/Object/floatobject.c.
+//
+// TODO(thschroeter):
+// - understand how it works
+// - bounds check for n
+// - test overflow
+func roundDigits(x float64, n int64) float64 {
+	var pow1, pow2, y, z float64
+	ndigits := float64(n)
+
+	if ndigits >= 0 {
+		if ndigits > 22 {
+			/* pow1 and pow2 are each safe from overflow, but
+			   pow1*pow2 ~= pow(10.0, ndigits) might overflow */
+			pow1 = math.Pow(10, ndigits-22)
+			pow2 = 1e22
+		} else {
+			pow1 = math.Pow(10, ndigits)
+			pow2 = 1.0
+		}
+		y = (x * pow1) * pow2
+		/* if y overflows, then rounded value is exactly x */
+		//if (!Py_IS_FINITE(y))
+		//    return PyFloat_FromDouble(x);
```

Comment:
To show the parts of the code that differ from the original. Will definitely remove these bits before merging, if this version will be used.